props/blaster.h

```diff
@@ -260,9 +233,92 @@ class Blaster : public PROP_INHERIT_PREFIX PropBase {
     }
   }
 
+  RefPtr<BufferedWavPlayer> auto_player_;
+
+  uint32_t auto_time_;
+  virtual void Fire() {
+    CheckEmpty();
+    if (empty_) return;
+#ifdef ENABLE_MOTION
+#ifdef BLASTER_JAM_PERCENTAGE
+    // If we're already jammed then we don't need to recheck. If we're not jammed then check if we just jammed.
+    is_jammed_ = is_jammed_ ? true : CheckJam(BLASTER_JAM_PERCENTAGE);
+    if (is_jammed_) {
+      SaberBase::DoEffect(EFFECT_JAM, 0);
+      return;
+    }
+#endif
+#endif
+    if (blaster_mode == MODE_AUTO) {
+      // Set up the autofire pairing if the font suits it.
+      SelectAutoFirePair();
+      auto_firing_ = true;
+      auto_time_ = millis();
+      SaberBase::SetLockup(LOCKUP_AUTOFIRE);
+      SaberBase::DoBeginLockup();
+      STDOUT << "***************************** OK Going to GetNewestAutoWav() with &SFX_auto in tow\n";
+      GetNewestAutoWav(&SFX_auto);
+      STDOUT << "***************************** Back here now and doing autofire like I should. \n";
+
+    } else {
+      if (blaster_mode == MODE_STUN) {
+        SaberBase::DoEffect(EFFECT_STUN, 0);
+        STDOUT << "******** STUN - Remaining shots = " << GetBulletCount() << "\n";
+      } else {
+        SaberBase::DoEffect(EFFECT_FIRE, 0);
+        STDOUT << "******** FIRE - Remaining shots = " << GetBulletCount() << "\n";
+      }
+      shots_fired_++;
+    }
+  }
+
+  RefPtr<BufferedWavPlayer> GetNewestAutoWav(Effect* a) {
```

Comment:
This is super confusing, because it looks like a "find a wav player and return it", but in fact it sets auto_player_ AND returns it, but then the return value is ignored where you call it.
I would recommend not setting auto_player_ inside this function, just return it and let the caller do the assignment.